patoolib/__init__.py

```diff
@@ -814,4 +814,31 @@ def get_inner_file_names_rar(file_path):
     'application/x-rar': get_inner_file_names_rar,
     'application/vnd.rar': get_inner_file_names_rar,
     'application/x-rar-compressed': get_inner_file_names_rar
-}
\ No newline at end of file
+}
+
+def get_inner_file_info(file_path):
+    files_list = []
+    if zipfile.is_zipfile(file_path):
+        return get_inner_file_info_zip(file_path, files_list)
+
+    if rarfile.is_rarfile(file_path):
+        return get_inner_file_info_rar(file_path, files_list)
+    # TODO: Add support to 7z archives using py7zr (python3)
+
+def get_inner_file_info_zip(file_path, files_list):
+    try:
+        with zipfile.ZipFile(file_path) as zf:
+            for info in zf.infolist():
+                files_list.append({"filename": info.filename,"file_size": info.file_size,"compress_size": info.compress_size})
+        return files_list
+    except Exception:
+        return []
```

Comment:
Return `files_list` instead of `[]`, because if the exception occurs in the 3rd iteration, you'll at least have already 2 items in the list.